### PR TITLE
Bump scarb version to 2.3.0

### PR DESCRIPTION
**Pull request by THenry14**
Closes https://github.com/foundry-rs/starknet-foundry/issues/429

## Introduced changes

- Bump scarb version for tests in cast to 2.3.0

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 #### Changed
 - bump starknet-rs dependency
 - replaced starknet-devnet with starknet-devnet-rs for tests
+- align scarb version in tests with forge
```
Comment (cptartur) [inline CHANGELOG.md:21]: Does it make sense to include it in changelong if it's only related to tests?

**Review by cptartur — APPROVED** [review]
🙏

**Review by piotmag769 — APPROVED** [review]
<3

File: CHANGELOG.md (hunk 1 above)
Comment (THenry14) [reply CHANGELOG.md]: Following discussin in https://github.com/foundry-rs/starknet-foundry/pull/812, removed this :+1:

**Pull request merged**